modules/cco-short-term-creds-auth-flow-aws.adoc

```diff
@@ -6,7 +6,49 @@
+
+AWS STS only grants access keys for requests that include service account tokens that meet the following conditions:
+
+* The token name and namespace match the service account name and namespace.
+
+* The token is signed by a key that matches the public key.
+
+The public key pair for the service account signing key used by the cluster is stored in an AWS S3 bucket. AWS STS federation validates that the service account token signature aligns with the public key stored in the S3 bucket.
+
+[id="cco-short-term-creds-auth-flow-aws-diagram_{context}"]
+== Authentication flow for AWS STS
+
+The following diagram illustrates the authentication flow between AWS and the {product-title} cluster when using AWS STS.
+
+* _Token signing_ is the Kubernetes service account signing service on the {product-title} cluster.
+* The _Kubernetes service account_ in the pod is the signed service account token.
 
 .AWS Security Token Service authentication flow
-image::347_OpenShift_credentials_with_STS_updates_0623_AWS.png[Detailed authentication flow between AWS and the cluster when using AWS STS]
\ No newline at end of file
+image::347_OpenShift_credentials_with_STS_updates_0623_AWS.png[Detailed authentication flow between AWS and the cluster when using AWS STS]
+
+Requests for new and refreshed credentials are automated by using an appropriately configured AWS IAM OpenID Connect (OIDC) identity provider, combined with AWS IAM roles. {product-title} signs service account tokens that are trusted by AWS IAM, and can be projected into a pod and used for authentication.
+
+[id="cco-short-term-creds-auth-flow-aws-refresh-policy_{context}"]
+== Token refreshing for AWS STS
+
+The signed service account token that a pod uses expires after a period of time. For clusters that use AWS STS, this time period is 3600 seconds, or one hour.
+
+The kubelet on the node that the pod is assigned to ensures that the token is refreshed. The kubelet attempts to rotate a token when it is older than 80 percent of its time to live.
+
+[id="cco-short-term-creds-auth-flow-aws-oidc_{context}"]
+== OpenID Connect requirements for AWS STS
+
+You can store the public portion of the encryption keys for your OIDC configuration in a public or private S3 bucket.
+
+The OIDC spec requires the use of HTTPS. AWS services require a public endpoint to expose the OIDC documents in the form of JSON web key set (JWKS) public keys. This allows AWS services to validate the bound tokens signed by Kubernetes and determine whether to trust certificates. As a result, both S3 bucket options require a public HTTPS endpoint and private endpoints are not supported.
+
+To use AWS STS, the public AWS backbone for the AWS STS service must be communicate with a public S3 bucket or a private S3 bucket with a public CloudFront endpoint. You can choose which type of bucket to use when you process `CredentialsRequest` objects during installation.
```

Comment:
Just a question/suggestion: if the second sentence is meant to introduce the list of options, should it end with a colon instead of a period?

```suggestion
To use AWS STS, the public AWS backbone for the AWS STS service must be communicate with a public S3 bucket or a private S3 bucket with a public CloudFront endpoint. You can choose which type of bucket to use when you process `CredentialsRequest` objects during installation:
```